sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveInlineTablesSuite.scala

```diff
@@ -117,8 +117,9 @@ class ResolveInlineTablesSuite extends AnalysisTest with BeforeAndAfter {
     val table = UnresolvedInlineTable(Seq("c1"),
       Seq(Seq(Cast(lit("1991-12-06 00:00:00.0"), TimestampType))))
     val withTimeZone = ResolveTimeZone.apply(table)
+    val resolvedStringTypes = ResolveDefaultStringTypes.apply(withTimeZone)
```

Comment:
why do we need this change? It's not under DML command and the session collation is not set, so the optimization should be triggered.

Reply:
makes sense, updated the logic in `needsResolution`